flybywiresim/aircraft
# feat(A380X/BAT): Move BAT to avionics framework

## Summary of Changes

Converts the BAT instrument component from React to the MSFS avionics framework. 

## Screenshots (if necessary)


Visually the component is identical to the previous implementation. 

![bat_sw](https://github.com/user-attachments/assets/cfe8b9af-2b3a-4947-a7e6-365a7820d74b)

## References

- Previous implementation
- Video of IRL usage




## Additional context

When moving the BAT-SW from OFF to one of the other positions, the value resets to '00.0V', before showing the actual voltage. Currently, the '00.0V' are set directly, as soon as the BAT-SW is turned to the OFF position. 


Discord username (if different from GitHub): nikolas_g

## Testing instructions

1. Load the A380X
2. Navigate to the upper overhead and move the switch position
3. Compare with LValues `L:A32NX_ELEC_BAT_{1,2,3,4}_POTENTIAL`


## How to download the PR for QA
- Simply build the BAT component (for example using `mach build -f BAT`

Every new commit to this PR will cause new A32NX and A380X artifacts to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, find and click on the **PR Build** tab
1. Click on either **flybywire-aircraft-a320-neo**, **flybywire-aircraft-a380-842 (4K)** or **flybywire-aircraft-a380-842 (8K)** download link at the bottom of the page
